src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Extensions/SumTypeExtensions.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Extensions
+{
+    internal static class SumTypeExtensions
```

Comment:
Should we make a type that inherits from `SumType<TextDocumentEdit, CreateFile, RenameFile, DeleteFile>` to make referencing it simpler / more understandable?

Reply:
It would certainly make some of these method definitions shorter lol, will do.

Reply:
Well, it would have been nice but apparently SumType is a `struct` and thus sealed. No inheritance for us.